addon/components/sl-date-time.js

```diff
@@ -154,8 +165,13 @@ export default Ember.Component.extend( TooltipEnabled, {
     momentValue: Ember.computed(
         'value',
         function() {
-            return window.moment( this.get( 'value' ) )
-                .locale( this.get( 'locale' ) );
+            const locale = this.get( 'locale' );
+
+            if ( locale ) {
+                return window.moment( this.get( 'value' ) ).locale( locale );
+            }
+
+            return null;
```

Comment:
`locale` (https://github.com/theoshu/sl-ember-components/blob/issue510/addon/components/sl-date-time.js#L67) and `value` (https://github.com/theoshu/sl-ember-components/blob/issue510/addon/components/sl-date-time.js#L90) are always set so this computed property always returns a value, negating the need to return a null.  These changes can be reverted.